shared/fs/common/path-item-action/menu-container.js

```diff
@@ -34,8 +36,13 @@ const mapDispatchToProps = (dispatch, {path, routePath}: OwnProps) => ({
   _confirmSendToOtherApp: () =>
     dispatch(FsGen.createSetPathItemActionMenuView({view: 'confirm-send-to-other-app'})),
   _copyPath: () => dispatch(ConfigGen.createCopyToClipboard({text: Constants.escapePath(path)})),
-  _delete: () => {
-    dispatch(FsGen.createDeleteFile({path}))
+  _deleteFile: () => {
+    dispatch(
+      RouteTreeGen.createNavigateTo({
+        parentPath: [fsTab],
+        path: [{props: {path}, selected: 'reallyDelete'}],
+      })
+    )
     dispatch(Constants.makeActionForOpenPathInFilesTab(Types.getPathParent(path), routePath))
```

Comment:
This should be moved to inside the confirm modal, after delete succeeds. Otherwise it'd navigate to the parent path even when delete doesn't happen.

Reply:
Hmm, I realized that I hadn't tested deleting the folder you're currently browsed to, and so when I moved this I tested it [before and after moving this line], and it errors the GUI out. It gets an error in an `OopsNoAccess`, which maybe it tries to make when the folder it's in stops existing. Any thoughts on how to avoid this?

Reply:
Good catch! There are two issues: 1) I missed the Oops (and another place) when doing the nav2 work. Sorry! We can't use `routeProps` directly in nav2. Here's a fix: `songgao/fix-nav2-routeProps` feel free to take it in. 2) It seems we need to navigateUp twice if we are inside the folder/file being deleted. This can be plumbed through routeProps when we push the navigationAppend from "Delete". As a result, path-item-action needs to know if it's part of a row or a header. So this would need some plumbing.

I guess it's fine to merge this if we put "delete" behind a feature flag. But if you want to address it in this PR before merge, I'm OK with that too.

Sorry for the bad estimate! This is definitely more than 2 points.

Reply:
I'll address it in this PR. Thanks for the helpful explanations!  🙏

Reply:
Now that (1) is fixed (thanks!), as soon as the folder you're in gets deleted, an `Oops` appears because the folder we're in doesn't exist. I can't figure out how to clear it from the `ReallyDelete`.

I've plumbed through to be able to tell if it's going to happen / if the pathItemAction is in the header.